src/Language/DifferentialDatalog/Rule.hs

```diff
@@ -222,3 +224,13 @@ ruleHeadIsRecursive d Rule{..} head_idx =
 ruleIsRecursive :: DatalogProgram -> Rule -> Bool
 ruleIsRecursive d rl@Rule{..} =
     any (ruleHeadIsRecursive d rl) [0 .. length ruleLHS - 1]
+
+-- | True iff the prefix of the rule of length 'n' produces a stream.
```

Comment:
```suggestion
-- | True if the prefix of the rule of length 'n' produces a stream.
```

Reply:
"iff" is a standard abbreviation for "if and only if"